Make location of IATI XML ZIP file configurable

simon-20 opened the pull request.

jarofgreen commented on docs/getting-started.rst:
```diff
@@ -37,3 +37,12 @@ from `the registry <https://iatiregistry.org/>`__, as well as `the latest codeli
     # download all XML in the registry
     iatikit.download.data()
+
+By default iatikit will download all the IATI data from the Code for IATI Data Download tool. However, you can customise this by creating an `iatikit.ini` file and using the following setting:
```
Where should this ini file be placed?

simon-20 replied:
Have added a commit explaining where `iatikit.ini` should be placed.

jarofgreen commented:
Looks good to me (tho can we have CHANGELOG entries?)

Thanks,

jarofgreen commented:
CHANGELOG added - this looks good to me now, thanks!

simon-20 commented:
> Thanks for this, @simon-20! I definitely like the idea of making the download location configurable – that’s a welcome addition.
> 
> I wonder if the new config setting could allow the user to specify the zip URL directly, rather than having to point to another file that contains the zip URL?
> 
> I realise iatikit currently has this redirection, but I guess it’s still possible for this setting to work differently? Would you be happy to make that change? Or I can merge this as is, and change it on `dev`.

Sure, I can change it. 

Would you want it to retain the current behaviour with the added redirect as a default though, so that it stays backwards compatible?

andylolz commented:
> Would you want it to retain the current behaviour with the added redirect as a default though, so that it stays backwards compatible?

Yes, exactly – the default would need to continue working like this.

jarofgreen commented:
In your last comment @andylolz can any new use of requests be followed by .raise_for_status() to check for non 200 responses?

(I have a PR incoming adding that in existing places after this one is sorted.)

andylolz commented:
@jarofgreen good point!

Though it’s not strictly a new use of requests, just a refactor. So I think it would be fine to fix it in your PR (rather than this one) but I don’t mind either way!

In any case, I’ll leave with @simon-20 to amend this PR.

andylolz reviewed (changes requested):
This is a nice feature!

If you could amend so that it’s a `zip_url` config variable, rather than `url_containing_zip_url` ([per this comment](https://github.com/codeforIATI/iatikit/pull/60#issuecomment-2960784945)) that would be fab.

simon-20 committed:
ci: run tests on Ubuntu 22.04

Fix test runner to Ubuntu 22.04 as ubuntu-latest is now Ubuntu 24.04 which doesn't support Python 3.7.

simon-20 committed:
feat: allow runtime configuration of IATI XML url

Python's config library doesn't allow `None` for default values so empty string is used as the default non-set value.

simon-20 committed:
doc: how to customise the source of XML

andylolz reviewed (approved):
Perfect – thank you!

andylolz commented:
This feature is available in v3.5.0.
